pkg/controller/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -315,6 +335,101 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileCredentialsRequest) usePassthroughCreds(cr *minterv1.CredentialsRequest, cloudCredSecret *corev1.Secret, logger log.FieldLogger) (*minterv1.CredentialsRequest, error) {
+	secretAlreadyExists := true
+	existingSecret := &corev1.Secret{}
+	err := r.Get(context.Background(), types.NamespacedName{Name: cr.Spec.SecretRef.Name, Namespace: cr.Spec.SecretRef.Namespace}, existingSecret)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Debug("existing secret not found. will create one.")
+			secretAlreadyExists = false
+		} else {
+			logger.WithError(err).Error("error checking if the secret satisfying the credentialsrequest already exists")
+			return nil, err
+		}
+	}
+
+	if secretAlreadyExists {
+		logger.Debug("Update if necessary")
+
+		if updateExistingSecret(existingSecret, cloudCredSecret) {
+			logger.Debugf("updating existing secret")
+
+			err = r.Update(context.Background(), existingSecret)
+			if err != nil {
+				logger.WithError(err).Error("error updating existing secret")
+				return nil, err
+			}
+		}
+	} else {
+		logger.Debug("Creating a secret")
+
+		newSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cr.Spec.SecretRef.Name,
+				Namespace: cr.Spec.SecretRef.Namespace,
+			},
+			Data: map[string][]byte{
+				secretannotator.AwsAccessKeyName:       cloudCredSecret.Data[secretannotator.AwsAccessKeyName],
+				secretannotator.AwsSecretAccessKeyName: cloudCredSecret.Data[secretannotator.AwsSecretAccessKeyName],
+			},
+		}
+
+		// set controller reference
+		if err := controllerutil.SetControllerReference(cr, newSecret, scheme.Scheme); err != nil {
+			logger.WithError(err).Error("error setting controller reference on secret")
+			return nil, err
+		}
+
+		// create secret
+		if err := r.Create(context.Background(), newSecret); err != nil {
```

Comment:
r.Client.Create I'm guessing.

I'm nervous that this is bypassing the actuator entirely though.

Reply:
r.Create works because we've dumped the kubeclient straight into the struct
```
type ReconcileCredentialsRequest struct {
	client.Client
	Actuator actuator.Actuator
}
```

Reply:
Aha I see, wasn't aware we could do that.